packages/systemd/prepare-var-lib-systemd-linger.service

```diff
@@ -0,0 +1,20 @@
+[Unit]
+Description=Prepare systemd-inhibit Directory (/var/lib/systemd/linger)
+DefaultDependencies=no
+RequiresMountsFor=/var
+RefuseManualStart=true
+RefuseManualStop=true
+
+[Service]
+Type=oneshot
+
+# Create /var/lib/systemd/linger so it is available for systemd-logind.
+ExecStart=/usr/bin/mkdir -p /var/lib/systemd/linger
```

Comment:
I'd prefer to do this in a tmpfiles snippet where we completely delete, then recreate and relabel the directory (like in `selinux-policy-tmpfiles.conf`).

But if we don't need to persist the data, having `systemd-inhibit` use `/run` instead would be better.

Reply:
I don't think we have much control over where `systemd-inhibit` wants to write to here. I could look in to patching the source to change its default, but there are other files written under `/var/lib/systemd` and we have tmpfile configuration for that.

The problem I ran in to was it appears to want to have the `linger` directory already exist under `/var/log/systemd` and doesn't appear to try to create it itself if it's not there. It just errors out and the `systemd-logind` service just fails to start.

If there's a way to have the directory created with the [tmpfile configuration](https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/systemd/systemd-tmpfiles.conf#L4), I can add it there. But I thought we've had to do it this way in the past since we need to make sure the tmpfile path is mounted first before we can make modifications inside it.

Reply:
Understood now, working on doing this all through the tmpfiles configuration.